ui/raidboss/data/05-shb/eureka/zadnor.js

```diff
@@ -81,30 +133,1155 @@ export default {
+      alertText: (data, _matches, output) => {
+        // TODO: how does this loop?
+        if (data.diabloSeenDealing)
+          return output.knockbackNox();
+        return output.knockbackBits();
+      },
+      run: (data) => data.diabloSeenDealing = true,
+      outputStrings: {
+        knockbackBits: {
+          en: 'Knockback (away from bits)',
+        },
+        knockbackNox: {
+          en: 'Knockback (into empty corner)',
+        },
+      },
+    },
+    {
+      id: 'Zadnor Diablo Void Systems Overload',
+      netRegex: NetRegexes.startsUsing({ source: 'The Diablo Armament', id: '5CB7', capture: false }),
+      response: Responses.bigAoe(),
+    },
+    {
+      id: 'Zadnor Diablo Pillar Of Shamash Spread',
+      // 5CBC damage
+      netRegex: NetRegexes.headMarker({ id: '0017' }),
+      condition: (data) => data.ce === 'dalriadaDiablo',
+      preRun: (data, matches) => {
+        data.diabloPillar = data.diabloPillar || [];
+        data.diabloPillar.push(matches.target);
+      },
+      alertText: (data, matches, output) => {
+        if (data.me === matches.target)
+          return output.text();
+      },
+      outputStrings: {
+        text: {
+          en: 'Laser Spread on YOU',
+        },
+      },
+    },
+    {
+      id: 'Zadnor Diablo Pillar Of Shamash Stack',
+      // 5CBE damage (no headmarker???)
+      netRegex: NetRegexes.headMarker({ id: '0017', capture: false }),
+      condition: (data) => data.ce === 'dalriadaDiablo',
+      delaySeconds: 1.5,
+      suppressSeconds: 5,
+      infoText: (data, _matches, output) => {
+        if (!data.diabloPillar || !data.diabloPillar.includes(data.me))
+          return output.text();
+      },
+      run: (data) => delete data.diabloPillar,
+      outputStrings: {
+        text: {
+          en: 'Line Stack',
+        },
+      },
+    },
+    {
+      id: 'Zadnor Diablo Acceleration Bomb Dodge',
+      netRegex: NetRegexes.gainsEffect({ effectId: 'A61' }),
+      condition: Conditions.targetIsYou(),
+      durationSeconds: (_data, matches) => parseFloat(matches.duration) - 4,
+      infoText: (_data, matches, output) => {
+        // Durations are 7 and 12.
+        const duration = parseFloat(matches.duration);
+        if (duration > 10)
+          return output.dodgeFirst();
+        return output.dodgeSecond();
+      },
+      outputStrings: {
+        dodgeFirst: {
+          en: '(Dodge -> Stop)',
+        },
+        dodgeSecond: {
+          en: '(Stop -> Dodge)',
+        },
+      },
+    },
+    {
+      id: 'Zadnor Diablo Acceleration Bomb Stop',
+      netRegex: NetRegexes.gainsEffect({ effectId: 'A61' }),
+      condition: Conditions.targetIsYou(),
+      // TODO: this could be better timed to be later for the dodge -> stop version and earlier
+      // for the stop -> dodge.
+      delaySeconds: (_data, matches) => parseFloat(matches.duration) - 3.5,
+      response: Responses.stopEverything(),
```

Comment:
This is defaulted to alarmText

Reply:
I thiiiiink this is ok? There aren't many alarms here, and stop is usually alarm.  Feel free to try this out and tell me if you think it should be changed after.